### ISSUE TITLE
Improvement: OpenKJ - Test DB paths upon launch - currently its freezing OKJ on launch

### ISSUE THREAD

**Issue by backstagekaraoke** [OpenKJ/OpenKJ#164]
Linux Mint 19
Kernels 4.15.47 & 5.0.*
OpenKJ 1.4*

I had an issue recently where OpenKJ wouldn't open ... it would start, launch 2 OpenKJ processes which thrashed the CPU usage for about 10 mins and then they quit themselves. I opened the openkj.sqlite file and everything seemed fine ... after some more checking, it turns out the issue was caused by a changed in the mount point for my second hard drive which holds all my backing music and karaoke tracks. This change seems to cause OpenKJ to get confused, thrash the CPU for 10mins and then exit.

Renaming or deleted the openkj.sqlite file, allows OpenKJ to load, just without any DB information, so you can just reimport etc. Obviously this is not going to be a common thing, however I think a check of the DB paths at launch would be a good addition.

Suggestion:

At launch test if the DB paths are valid, if not, present either a warning advising that OpenKJ cannot open as it cannot locate the DB paths etc .... This would at least prevent OpenKJ for freezing for 10mins and also prompt the user to go investigate etc. 

In addition to the check, if the paths are valid, continue to load and then in the background you could initiate a check to see if any new files have been added to the source folders and present the user with an option to add them and update the local and remote DB.

**Comment by coyote1357** [comment]
When you triggered this, did you have the option to watch source dirs for new files turned on?

**Comment by backstagekaraoke** [comment]
No, all 3 options under the Database tab are unticked.

**Comment by coyote1357** [comment]
I'll have to see what is trying to access the database disk paths on startup, that's not supposed to happen until you actually try to play a file or update the database unless lazy duration loading or directory watches are enabled.  Something might be defaulting to on even when its off, which is less than ideal. heh

**Comment by backstagekaraoke** [comment]
ok let me know if there is anything you want me to check on my side.

**Comment by coyote1357** [comment]
Should be fixed in 1.5.51 unstable

**Comment by backstagekaraoke** [comment]
what was the issue? do you want me to replicate the missing paths and test on my side? or you confident with the fix?

**Comment by backstagekaraoke** [comment]
@coyote1357 ok replicated the missing drive/path, and unfortunately in 1.5.51 the same issue is happening. Locks up the launch for a few mins, then the process ends.

reinstate the drive/path and it loads as per normal.

**Comment by coyote1357** [comment]
I found a combination of things that could have been the reason for the hang.  One had already been fixed in unstable but not in stable, it was trying to create filesystem watches for directories even if the feature was disabled.  One of the new changes also checks to make sure that the directory exists and is readable if the directory watching is enabled, and only tries to add the watch if it's valid.  The other new change prevents it from trying to lazy load durations erroneously if the feature is disabled.

And damn, I thought one of those would be the root cause.  I'll keep looking.  If you could launch the program from the command line and capture the output for me, that might lead me in the right direction.

**Comment by coyote1357** [comment]
I noticed in your debug output from the other bug report that you appear to have break music set to start on program start.  If so, can you disable that feature and see if this still occurs?

**Comment by backstagekaraoke** [comment]
@coyote1357 before I do that here is the output replicating the missing paths(s):

[OpenKJCommandLineOutput.pdf](https://github.com/OpenKJ/OpenKJ/files/3133668/OpenKJCommandLineOutput.pdf)

**Comment by backstagekaraoke** [comment]
@coyote1357 ok changed the break music to not start at launch and OpenKJ loaded up even with the missing drive .... However I noticed after about 20 seconds my CPU fans started to go full blast .... I exited OpenKJ and they stopped. Then I launched from command line and got the "another instance running" error .... here is the output:

backstagekaraoke@BKLaptop02:~$ OpenKJ 
qt5ct: using qt5ct plugin
qt5ct: D-Bus global menu: no
Initial volumes - K:  99  BM:  51
Database schema version:  105
Creating dbModel
Creating queueModel
Singer selection is none
Creating qDelegate
creating rotModel
Creating dbDialog
Creating dlgKeyChange
Creating regularSingersDialog
Creating regularExportDialog
Creating regularImportDialog
Creating requestsDialog
tsWidth =  106
singerColSize =  101
Creating dlgBookCreator
QMetaObject::connectSlotsByName: No matching signal for on_comboBoxSort_currentIndexChanged(int)
Header font:  "Noto Sans,9,-1,5,50,0,0,0,0,0"
Item font:    "Noto Sans,9,-1,5,50,0,0,0,0,0"
Creating dlgEq
Creating dlgAddSinger
Creating dlgSongShop
Requesting songs from db.openkj.org
Creating CDG object
Creating dbDelegate
Creating break music audio backend object
Start constructing GStreamer backend
"BM"  - Initializing gst

0:00:00.036065728  3762 0x7fd440004d60 WARN     GST_ELEMENT_FACTORY gstelementfactory.c:456:gst_element_factory_make: no such element factory "ladspa-ladspa-rubberband-so-rubberband-pitchshifter-stereo"!
"BM"  - setVolume - setting to linear:  0
"BM"  - Setting output device to device idx:  0
"BM"  - Default device selected
"BM"  - AudioBackendGstreamer::setDownmix( false ) called
"BM"  - AudioBackendGstreamer::setDownmix() completed
Done constructing GStreamer backend
Creating karaoke audio backend object
Start constructing GStreamer backend
"KA"  - Initializing gst

0:00:00.063242369  3762 0x7fd440004d60 WARN     GST_ELEMENT_FACTORY gstelementfactory.c:456:gst_element_factory_make: no such element factory "ladspa-ladspa-rubberband-so-rubberband-pitchshifter-stereo"!
"KA"  - Pitch shifter SoundTouch enabled
"KA"  - setVolume - setting to linear:  0
"KA"  - Setting output device to device idx:  0
"KA"  - Default device selected
"KA"  - AudioBackendGstreamer::setDownmix( false ) called
"KA"  - AudioBackendGstreamer::setDownmix() completed
Done constructing GStreamer backend
Start constructing GStreamer backend
"SFX"  - Initializing gst

0:00:00.082958974  3762 0x7fd440004d60 WARN     GST_ELEMENT_FACTORY gstelementfactory.c:456:gst_element_factory_make: no such element factory "ladspa-ladspa-rubberband-so-rubberband-pitchshifter-stereo"!
"SFX"  - setVolume - setting to linear:  0
"SFX"  - Setting output device to device idx:  0
"SFX"  - Default device selected
"SFX"  - AudioBackendGstreamer::setDownmix( false ) called
"SFX"  - AudioBackendGstreamer::setDownmix() completed
Done constructing GStreamer backend
Creating audio recorder object
AudioRecorder constructor called
AudioRecorder::initGStreamer() called
AudioRecorder::generateDeviceList() called
AudioRecorder - Initializing gstreamer
AudioRecorder - Creating elements
Elements created, adding to pipeline and linking
AudioRecorder::initGStreamer() completed
Creating settings dialog
AudioRecorder constructor called
AudioRecorder::initGStreamer() called
AudioRecorder::generateDeviceList() called
AudioRecorder - Initializing gstreamer
AudioRecorder - Creating elements
Elements created, adding to pipeline and linking
AudioRecorder::initGStreamer() completed
defaultServiceProvider::requestService(): no service found for - "org.qt-project.qt.audiosource"
Creating CDG output dialog
DlgCdg::setShowBgImage( true ) called
DlgCdg::setShowBgImage( true ) called
QObject::connect: No such slot DlgCdg::triggerBg(AbstractAudioBackend::State)
QObject::connect:  (receiver name: 'DlgCdg')
"KA"  - AudioBackendGstreamer::setDownmix( true ) called
"KA"  - AudioBackendGstreamer::setDownmix() completed
"BM"  - AudioBackendGstreamer::setDownmix( true ) called
"BM"  - AudioBackendGstreamer::setDownmix() completed
Setting backgrounds on CDG displays
DlgCdg::setShowBgImage( true ) called
Restoring window and listview states
DlgCdg::setShowBgImage( true ) called
Refreshing rotation data
Adjusting karaoke listviews column visibility and state
Adding singer count to status bar
Connecting signals & slots
Setting up volume sliders
"BM"  - setVolume - setting to linear:  0.132651
"KA"  - setVolume - setting to linear:  0.970299
Restoring multiplex button states
Creating karaoke autoplay timer
Requesting current version info for branch:  "unstable"
Initial UI stup complete
qt5ct: palette support is disabled
0:00:00.569890504  3762 0x7fd440004d60 FIXME                    bin gstbin.c:4336:gst_bin_query: implement duration caching in GstBin again
Resizing cdgVideoWidget to width:  300
SfxEntryList size:  0
DlgCdg::setShowBgImage( true ) called
Received network reply
Received version:  "1.5.33"  Current version:  "1.5.51"
QJsonDocument({"command":"getEntitledSystemCount","count":2,"error":"false"})
OKJSongbookAPI: Server reports entitled to run  2  concurrent systems
QJsonDocument({"command":"getEntitledSystemCount","count":2,"error":"false"})
OKJSongbookAPI: Server reports entitled to run  2  concurrent systems
Got network reply from db.openkj.org
LazyDurationUpdateController stoWork() called
Saving volumes - K:  99  BM:  51
QFileSystemWatcher::removePaths: list is empty
